### PR TITLE
fix: fix io handling in structural_simplify, input_output_handling tests

### PR DESCRIPTION
**Pull request by AayushSabharwal**
## Checklist

- [ ] Appropriate tests were added
- [ ] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API

### CODE REVIEW

**Review by baggepinnen — APPROVED** [review]

**Comment by AayushSabharwal** [comment]
Doc failure is just codecov being codecov

File: src/systems/diffeqs/abstractodesystem.jl [hunk 1]
```diff
@@ -875,7 +875,11 @@ function process_DEProblem(constructor, sys::AbstractODESystem, u0map, parammap;
         initializeprobmap = getu(initializeprob, unknowns(sys))
 
         zerovars = setdiff(unknowns(sys), keys(defaults(sys))) .=> 0.0
-        trueinit = identity.([zerovars; u0map])
+        if u0map isa Dict
+            trueinit = identity.([zerovars; collect(pairs(u0map))])
+        else
+            trueinit = identity.([zerovars; u0map])
+        end
```
Comment (ChrisRackauckas) [inline src/systems/diffeqs/abstractodesystem.jl:882]: Shouldn't this just be `trueinit = merge(zerovars, todict(u0map))`?

Comment (AayushSabharwal) [reply src/systems/diffeqs/abstractodesystem.jl]: Fixed

**Comment by ChrisRackauckas** [comment]
Tests fail

**Comment by AayushSabharwal** [comment]
Tests are fine now, it's just the Optimization thing that's new

**Pull request merged**